homeassistant/components/unifiprotect/sensor.py

```diff
@@ -0,0 +1,443 @@
+from datetime import datetime, timedelta
+import logging
+from typing import Any
+
+from pyunifiprotect.data import NVR
+from pyunifiprotect.data.base import ProtectAdoptableDeviceModel
+
+from homeassistant.components.sensor import (
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    DATA_BYTES,
+    DATA_RATE_BYTES_PER_SECOND,
+    DATA_RATE_MEGABITS_PER_SECOND,
+    DEVICE_CLASS_BATTERY,
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_ILLUMINANCE,
+    DEVICE_CLASS_SIGNAL_STRENGTH,
+    DEVICE_CLASS_TEMPERATURE,
+    DEVICE_CLASS_TIMESTAMP,
+    DEVICE_CLASS_VOLTAGE,
+    ELECTRIC_POTENTIAL_VOLT,
+    LIGHT_LUX,
+    PERCENTAGE,
+    SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+    TEMP_CELSIUS,
+    TIME_SECONDS,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import Entity, EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .data import ProtectData
+from .entity import ProtectDeviceEntity, ProtectNVREntity, async_all_device_entities
+from .models import ProtectRequiredKeysMixin
+from .utils import get_nested_attr
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class ProtectSensorEntityDescription(ProtectRequiredKeysMixin, SensorEntityDescription):
+    """Describes UniFi Protect Sensor entity."""
+
+    precision: int | None = None
+
+
+_KEY_UPTIME = "uptime"
+_KEY_BLE = "ble_signal"
+_KEY_WIRED = "phy_rate"
+_KEY_WIFI = "wifi_signal"
+
+_KEY_RX = "stats_rx"
+_KEY_TX = "stats_tx"
+_KEY_OLDEST = "oldest_recording"
+_KEY_USED = "storage_used"
+_KEY_WRITE_RATE = "write_rate"
+_KEY_VOLTAGE = "voltage"
+
+_KEY_BATTERY = "battery_level"
+_KEY_LIGHT = "light_level"
+_KEY_HUMIDITY = "humidity_level"
+_KEY_TEMP = "temperature_level"
+
+_KEY_CPU = "cpu_utilization"
+_KEY_CPU_TEMP = "cpu_temperature"
+_KEY_MEMORY = "memory_utilization"
+_KEY_DISK = "storage_utilization"
+_KEY_RECORD_ROTATE = "record_rotating"
+_KEY_RECORD_TIMELAPSE = "record_timelapse"
+_KEY_RECORD_DETECTIONS = "record_detections"
+_KEY_RES_HD = "resolution_HD"
+_KEY_RES_4K = "resolution_4K"
+_KEY_RES_FREE = "resolution_free"
+_KEY_CAPACITY = "record_capacity"
+
+ALL_DEVICES_SENSORS: tuple[ProtectSensorEntityDescription, ...] = (
+    ProtectSensorEntityDescription(
+        key=_KEY_UPTIME,
+        name="Uptime",
+        icon="mdi:clock",
+        device_class=DEVICE_CLASS_TIMESTAMP,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        ufp_value="up_since",
+    ),
+    ProtectSensorEntityDescription(
+        key=_KEY_BLE,
+        name="Bluetooth Signal Strength",
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+        device_class=DEVICE_CLASS_SIGNAL_STRENGTH,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
```

Comment:
Signal strength is one of the few cases we require to be disabled by default (as documented in the developer docs).